src/MongoDB/BulkWrite.c

```diff
@@ -31,6 +31,20 @@
 
 zend_class_entry *php_phongo_bulkwrite_ce;
 
+/* Returns whether the insert document appears to be a legacy index. */
+static inline bool php_phongo_bulkwrite_insert_is_legacy_index(bson_t *bdocument) /* {{{ */
```

Comment:
It proved much easier to inspect the BSON document than the original zval argument. The only reason I considered checking the zval was because we might want to disable `_id` generation for these documents, but pre-2.6 servers don't seem to care (the `_id` field never shows up in `system.indexes` document). Also, we would fail to properly generate an `_id` if this function detected a false positive (`bulkwrite-insert-003.phpt` tests for this).